R/python.R

```diff
@@ -1222,8 +1222,11 @@ py_eval <- function(code, convert = TRUE) {
 }
 
 py_callable_as_function <- function(callable, convert) {
-  function(...) {
-    dots <- py_resolve_dots(list(...))
+  f <- function(...) {
+    # Cannot use list(...), as we replace the formals() later.
+    call <- sys.call()
+    call[[1]] <- as.name('list')
```

Comment:
I've typically seen `alist()` used rather than `list()` for this sort of purpose. Would that work here?

Reply:
No, as `alist` exists so people can manually create a structure similar to what `sys.call` returns.

`sys.call` returns an unevaluated `call` expression and we want to evaluate things in the end. What `sys.call` returns in my implementation when we call `py_func(a, b = c, d = 1)` is:

```r
structure(
  list(
    as.name('py_func'),
    as.name('a'),
    b = as.name('c'),
    d = 1
  ),
  class = 'call'
)
```

I then replace `py_func` with `list` and evaluate the thing in the right context.

What `alist` does is simply explained by its definition: `function(...) as.list(sys.call())[-1L]`. It does nothing that isn’t already done at that point.

I hope this explains it. I rephrased this far too many times, so if it doesn’t, I fear you have to dive into this yourself, it’s a bit mind-bending and hard to phrase.